types/validation.go

```diff
@@ -97,16 +122,35 @@ func (c *Commit) ValidateBasic() error {
 
 func (c *Commit) ValidateWithHeader(proposerPubKey tmcrypto.PubKey, header *Header) error {
 	if err := c.ValidateBasic(); err != nil {
-		return err
+		return NewErrInvalidSignatureFraud(err)
+	}
+
+	if c.Height != header.Height {
+		return NewErrInvalidBlockHeightFraud(c.Height, header.Height)
+	}
```

Comment:
invalid block height fraud is a bit undefined IMO
this is an invalid commit fraud